target/linux/ath79/dts/ar7100_mikrotik_routerboard-rb4xx.dtsi

```diff
@@ -0,0 +1,154 @@
+// SPDX-License-Identifier: GPL-1.0-or-later OR MIT
+
+// Based on ar71xx arch/mips/ath79/mach-rb4xx.c
+
+/dts-v1/;
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+#include "ar7100.dtsi"
+
+/ {
+	chosen {
+		bootargs = "console=ttyS0,115200";
+	};
+
+	aliases {
+		led-boot = &led_user_yellow;
+		led-failsafe = &led_user_yellow;
+		led-running = &led_user_yellow;
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		led_user_yellow: user_yellow {
+			label = "rb4xx:yellow:user";
+			gpios = <&gpio 4 GPIO_ACTIVE_HIGH>;
+			default-state = "on";
+		};
+
+		led_1_green: 1_green {
+			label = "rb4xx:green:led1";
+			gpios = <&cpld 0 GPIO_ACTIVE_LOW>;
+		};
+
+		led_2_green: 2_green {
+			label = "rb4xx:green:led2";
+			gpios = <&cpld 2 GPIO_ACTIVE_LOW>;
+		};
+
+		led_3_green: 3_green {
+			label = "rb4xx:green:led3";
+			gpios = <&cpld 1 GPIO_ACTIVE_LOW>;
+		};
+
+		led_4_green: 4_green {
+			label = "rb4xx:green:led4";
+			gpios = <&cpld 3 GPIO_ACTIVE_LOW>;
+		};
+
+		led_5_green: 5_green {
+			label = "rb4xx:green:led5";
+			gpios = <&cpld 8 GPIO_ACTIVE_HIGH>;
+		};
+	};
+
+	keys: keys {
+		compatible = "gpio-keys";
+
+		status = "disabled";
```

Comment:
What's that supposed to do?

Reply:
So this is an interesting one. All the other rb4xx boards (that would derive from this dtsi) have a working reset button active on gpio 7. There's one exception - the rb493g shares this gpio with the mdio-gpio "MDC" and so it doesn't work for all subsets, and so I've marked it disabled by default.

(I've read it's not possible to model this gpio "sharing" in Linux, and the old mach-rb4xx.c file has a similar configuration too)

Reply:
I do not have time to look into this right now, but I have never seen somebody using status for that. I suspect you just want to have that button defined for some devices, but not all, so please move it to the DTSes of these files. The shared DTSI should only contain what's there for all devices.

Reply:
The rb433, 435g, 450, 450g and 493 all have this reset key (and those 5 boards would be children of this parent 4xx .dtsi file). The rb493g I have is the only child that doesn't have that key. (i.e. 5 out of the 6 have it.)

I can see two options:

(1) This "keys" section could be copied into those 5 future .dts files if/when those boards are ported and be deleted from the .dtsi here.

(2) I could set it active in the .dtsi and mark it disabled in the 1 child (the 493g) that doesn't have it. That would result in less duplication.

I'm happy to take your recommendation here.

Reply:
Either do (1) or create a second derived DTSI for the four devices if it's worth it (just having the keys there means it's not worth it).

Reply:
I will go with #1 and delete this disabled section for now.